### PR TITLE
Added support for non-csv output of XMatch

### PR DESCRIPTION
**Pull request by minzastro**
Added support for "responseformat" and "selection" parameters for XMatch requests.

### CODE REVIEW

File: astroquery/xmatch/core.py [hunk 1]
```diff
@@ -57,15 +58,21 @@ def query(self, cat1, cat2, max_distance, colRA1=None, colDec1=None,
...
                                     colRA2, colDec2, cache=cache,
-                                    get_query_payload=get_query_payload)
+                                    get_query_payload=get_query_payload,
+                                    responseformat=responseformat,
+                                    selection=selection)
         if get_query_payload:
             return response
-        return ascii.read(response.text, format='csv')
+        fp = six.BytesIO()
+        fp.write(response.text.encode('ascii'))
```
Comment (keflavich) [inline astroquery/xmatch/core.py]: Is this equivalent to `response.content`?

**Comment by keflavich** [comment]
@minzastro Do you think you could add a test for this change, or at least add an example so we can figure out how to write an appropriate test?

**Comment by bsipocz** [comment]
@minzastro - I acknowledge it was a while ago, but could you please elaborate a bit why the response format matter when it's purely internal detail and the users get an astropy Table anyway?

Some context: we're planning to switch away from parsing the csv response in favour of votable.

**Comment by minzastro** [comment]
> @minzastro - I acknowledge it was a while ago, but could you please elaborate a bit why the response format matter when it's purely internal detail and the users get an astropy Table anyway?
> 
> Some context: we're planning to switch away from parsing the csv response in favour of votable.
There was a dual goal. First, I wanted to have an option to receive VOTable in query_async, i.e. without parsing to Table object - is it not always necessary. Second, I also added a 'selection' parameter to both query_async and query, as the default 'selection="all"' is not always the desired choice.
Sorry, I never had the time/motivation to complete this push request.

**Pull request closed without merging**